compiler/optimizer/VPHandlers.cpp

```diff
@@ -2387,16 +2383,13 @@ TR::Node *constrainIaload(OMR::ValuePropagation *vp, TR::Node *node)
                   {
                   TR::VMAccessCriticalSection constrainIaloadCriticalSection(vp->comp(),
                                                                               TR::VMAccessCriticalSection::tryToAcquireVMAccess);
-                  void * p = symbol->getStaticAddress();
-                  if ((*(void **)p != 0) &&
+
+                  uintptrj_t arrayStaticAddress = (uintptrj_t)symbol->getStaticAddress();
+                  uintptrj_t arrayObject = TR::Compiler->om.getStaticReferenceFieldAtAddress(vp->comp(), arrayStaticAddress, symRef);
+                  if ((arrayObject != 0) &&
                       constrainIaloadCriticalSection.hasVMAccess())
                      {
-                     arrLength = *((int32_t *) (((uintptrj_t) *(void ***)p) + (uintptrj_t) vp->fe()->getOffsetOfContiguousArraySizeField()));
-
-                     if (arrLength == 0 && TR::Compiler->om.useHybridArraylets())
-                        {
-                        arrLength = *((int32_t *) (((uintptrj_t) *(void ***)p) + (uintptrj_t) vp->fe()->getOffsetOfDiscontiguousArraySizeField()));
-                        }
+                     //int32_t arrLength = TR::Compiler->om.getArrayLengthInElements(vp->comp(), arrayObject);
```

Comment:
Is this line meant to be commented? Can it be removed?

Reply:
I kept it commented because of 

https://github.com/eclipse/omr/pull/2344/files/0b3b80243b9c67f50c89fc700e308cdb12d67ffc#diff-30d6231397753ecc45846719f86be793R2401 and https://github.com/eclipse/omr/pull/2344/files/0b3b80243b9c67f50c89fc700e308cdb12d67ffc#diff-30d6231397753ecc45846719f86be793R2402

Reply:
Could you please add a comment above the commented out line saying why it is there so it is not removed by subsequent cleanup.